x/tokenomics/keeper/token_logic_modules.go

```diff
@@ -0,0 +1,443 @@
+package keeper
+
+// References:
+// - https://docs.pokt.network/pokt-protocol/the-shannon-upgrade/proposed-tokenomics/token-logic-modules
+// - https://github.com/pokt-network/shannon-tokenomics-static-tests
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"cosmossdk.io/math"
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/pokt-network/smt"
+
+	"github.com/pokt-network/poktroll/app/volatile"
+	"github.com/pokt-network/poktroll/pkg/crypto/protocol"
+	"github.com/pokt-network/poktroll/telemetry"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	prooftypes "github.com/pokt-network/poktroll/x/proof/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
+	tokenomicstypes "github.com/pokt-network/poktroll/x/tokenomics/types"
+	tokenomictypes "github.com/pokt-network/poktroll/x/tokenomics/types"
+)
+
+const (
+	// TODO_UPNEXT(@olshansk): Make all of the governance params
+	MintAllocationDAO         = 0.1
+	MintAllocationProposer    = 0.05
+	MintAllocationSupplier    = 0.7
+	MintAllocationSourceOwner = 0.15
+	MintAllocationApplication = 0.0
+	// TODO_UPNEXT(@olshansk): Remove this. An ephemeral placeholder before
+	// real values are introduced.
+	MintGlobalAllocation = 0.0000000
+)
+
+type TokenLogicModule int
+
+func (tlm TokenLogicModule) String() string {
+	return [...]string{
+		"TLMRelayBurnEqualsMint",
+		"TLMGlobalMint",
+	}[tlm]
+}
+
+func (tlm TokenLogicModule) EnumIndex() int {
+	return int(tlm)
+}
+
+const (
+	TLMRelayBurnEqualsMint TokenLogicModule = iota
+	TLMGlobalMint
+	// TODO_UPNEXT(@olshansk): Add more TLMs
+)
+
+// TokenLogicModuleProcessor is the method signature that all token logic modules
+// are expected to implement.
+type TokenLogicModuleProcessor func(
```

Comment:
Could you please add a comment that these processors do update the target `Application` and `Supplier` but do not persist them?

The signature may mean the opposite.

Also, it seems to me that not all TLM processors need to update the `Application` record. It's maybe preferable to have it updated within the processor.